divide_and_conquer/closest_pair_of_points.py

```diff
@@ -79,35 +78,41 @@ def closest_pair_of_points_sqr(points, points_counts):
 
     # base case
     if points_counts <= 3:
-        return dis_between_closest_pair(points, points_counts)
+        return dis_between_closest_pair(points_sorted_on_X, points_counts)
     
     # recursion
     mid = points_counts//2
-    closest_in_left = closest_pair_of_points(points[:mid], mid)
-    closest_in_right = closest_pair_of_points(points[mid:], points_counts - mid)
+    closest_in_left = closest_pair_of_points_sqr(points_sorted_on_X, 
+                                                 points_sorted_on_Y[:mid], 
+                                                 mid)
+    closest_in_right = closest_pair_of_points_sqr(points_sorted_on_Y, 
+                                                  points_sorted_on_Y[mid:], 
+                                                  points_counts - mid)
     closest_pair_dis = min(closest_in_left, closest_in_right)
     
     """ cross_strip contains the points, whose Xcoords are at a 
     distance(< closest_pair_dis) from mid's Xcoord
     """
 
     cross_strip = []
-    for point in points:
-        if abs(point[0] - points[mid][0]) < closest_pair_dis:
+    for point in points_sorted_on_X:
+        if abs(point[0] - points_sorted_on_X[mid][0]) < closest_pair_dis:
             cross_strip.append(point)
 
-    cross_strip = column_based_sort(cross_strip, 1)
     closest_in_strip = dis_between_closest_in_strip(cross_strip, 
                      len(cross_strip), closest_pair_dis)
     return min(closest_pair_dis, closest_in_strip)
 
     
 def closest_pair_of_points(points, points_counts):
-    return math.sqrt(closest_pair_of_points_sqr(points, points_counts))
+    points_sorted_on_X = column_based_sort(points, column = 0)
+    points_sorted_on_Y = column_based_sort(points, column = 1)
+    return (closest_pair_of_points_sqr(points_sorted_on_X, 
+                                       points_sorted_on_Y, 
+                                       points_counts)) ** 0.5
 
 
-points = [(2, 3), (12, 30), (40, 50), (5, 1), (12, 10), (0, 2), (5, 6), (1, 2)]
-points = column_based_sort(points)
+points = [(2, 3), (12, 30), (40, 50), (5, 1), (12, 10), (3, 4)] 
```

Comment:
Please consider adding __if \_\_name\_\_ == "__main\_\_" before this line and indent this line and any lines below it.  This will allow us to run doctests without executing your code.

Reply:
ok.